shuup/front/apps/customer_information/dashboard_items.py

```diff
@@ -35,8 +36,9 @@ def show_on_dashboard(self):
         return False  # Don't show this on menu since we already have all the information there
 
     def show_on_menu(self):
-        # Only show this on menu if customer is company
-        return isinstance(self.request.customer, CompanyContact)
+        # Only show this on menu if customer is company and if company registration is allowed
+        is_registration_allowed = allow_company_registration(self.request.shop)
+        return (isinstance(self.request.customer, CompanyContact) and is_registration_allowed)
```

Comment:
I don't get it.. If the contact is a already a company, this shouldn't be an issue. Imagine that the shop has this option enabled and suddenly the merchant disables it, what happens to company contacts that we already created? I guess this is not an issue.